Mpfz default constructor causes unreachable code warning is Visual Studio

## Issue Details

Please see the below minimal code to reproduce the problem:

CMakeLists.txt
```
cmake_minimum_required(VERSION 3.11)

set(CMAKE_TOOLCHAIN_FILE
    $ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake
    CACHE STRING "")
set(VCPKG_TARGET_TRIPLET $ENV{VCPKG_DEFAULT_TRIPLET} CACHE STRING "")

project(Example)

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX /W4")

set(PROJECT_SRCS ${PROJECT_SOURCE_DIR}/main.cpp)

find_package(CGAL REQUIRED)

add_executable(${PROJECT_NAME} ${PROJECT_SRCS})

target_link_libraries(${PROJECT_NAME} CGAL::CGAL)
```

main.cpp
```
__pragma(warning(push,0))
#include "CGAL/Exact_predicates_inexact_constructions_kernel.h"
#include "CGAL/Polygon_mesh_processing/remesh.h"
#include "CGAL/Surface_mesh.h"
__pragma(warning(pop))

int main(){
  using Pt_3 = CGAL::Exact_predicates_inexact_constructions_kernel::Point_3;
  using Mesh = CGAL::Surface_mesh<Pt_3>;

  Mesh m;

  CGAL::Polygon_mesh_processing::isotropic_remeshing(m.faces(), 5, m);

  return 0;
}
```

Error message
```
1>main.cpp
1>c:\dev\vcpkg\installed\x64-windows\include\cgal\mpzf.h(273): error C2220: warning treated as error - no 'object' file generated
1>c:\dev\vcpkg\installed\x64-windows\include\cgal\mpzf.h(273): warning C4702: unreachable code
1>c:\dev\vcpkg\installed\x64-windows\include\cgal\mpzf.h(273): warning C4702: unreachable code
1>c:\dev\vcpkg\installed\x64-windows\include\cgal\mpzf.h(273): warning C4702: unreachable code
1>Done building project "Example.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 1 up-to-date, 0 skipped ==========
```

line 273 of mpzf.h is
```
inline mp_limb_t*& data() { return data_; }
```

If I comment out `isotropic_remeshing`, the error disappears. My guess is `isotropic_remeshing` indirectly uses something in mpzf.h that causes the warning to trigger, because if I include `CGAL/mpzf.h` directly without calling `isotropic_remeshing`, it actually compiles. The weird thing is this warning is not suppressed by the pragma warning push pop calls that surrounded the CGAL headers, so I don't even know how to temporarily work around this problem. Please help. 

## Environment

* Operating system: Windows 10
* Compiler: Visual Studio 2017
* Release or debug mode: Release
* CGAL version: 4.14 from `vcpkg install cgal[*]`
* Boost version: 1.71

--- a/Number_types/include/CGAL/Mpzf.h
+++ b/Number_types/include/CGAL/Mpzf.h
@@ -86,10 +86,11 @@
 
 #if defined(BOOST_MSVC)
 #  pragma warning(push)
-#  pragma warning(disable:4146 4244 4267 4800)
+#  pragma warning(disable:4146 4244 4267 4702 4800)
      // warning on - applied on unsigned number
      // conversion with loss of data
      // conversion with loss of data
+     // unreachable code
      // int to bool performance
 #endif
 

